qa/rpc-tests/sc_big_commitment_tree.py

```diff
@@ -0,0 +1,254 @@
+        test_helper.create_sidechain("test_sidechain", 2)
+        sc_id = test_helper.get_sidechain_id("test_sidechain")
+        mark_logs(f"Sidechain {sc_id} has been created", self.nodes, DEBUG_MODE, 'e')
+
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # Send some coins to Node1
+        # By using a new address for Node1, we are sure that any transaction created by Node1
+        # will use the change of the previous one, forcing the getblocktemplate() function
+        # to sort transaction based on submission order (because any transaction will depend
+        # on the previous one)
+        node1_address = self.nodes[1].getnewaddress()
+        self.nodes[0].sendtoaddress(node1_address, 100)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        current_sc_txs_commitment_tree_root = ""
+
+        forwardTransferOuts = [{'toaddress': "aabb", 'amount': Decimal(0.001), "scid": sc_id, "mcReturnAddress": node1_address}]
+
+        # Add to mempool 4 transactions, each containing 1024, 1024, 1024 and 1022 forward transfers (total 4094,
+        # which is almost the limit size of ScTxsCommTree)
+        mark_logs(f"Adding 4094 FT for sc {sc_id} to the mempool (4 transactions)", self.nodes, DEBUG_MODE, 'g')
+        multipll = [1024, 1024, 1024, 1022]
+        for i in range(4):
+            self.nodes[1].sc_send(forwardTransferOuts * multipll[i], {"fee": Decimal(0.0001 * (i + 1)), "minconf": 0})
+            mark_logs(f"Sent transaction #{i + 1}", self.nodes, DEBUG_MODE, 'c')
+            self.sync_all()
+
+            block_template = self.wait_for_block_template(0, i + 1)
+            assert(block_template is not None)
+            mark_logs(f"getblocktemplate() ->", self.nodes, DEBUG_MODE, 'e')
+            for k, transaction in enumerate(block_template['transactions']):
+                mark_logs(f"  {k} - {transaction['hash']}", self.nodes, DEBUG_MODE, 'e')
+            mark_logs(f"    ScTxsCommitmentTree root: {block_template['scTxsCommitment']}", self.nodes, DEBUG_MODE, 'e')
+
+            assert_false(current_sc_txs_commitment_tree_root == block_template['scTxsCommitment'])
+            current_sc_txs_commitment_tree_root = block_template['scTxsCommitment']
+
+
+        # Let's try to add two more forward transfers to the mempool to exceed ScTxsCommTree capacity
+        mark_logs(f"Adding 2 FT for sc {sc_id} to the mempool", self.nodes, DEBUG_MODE, 'y')
+        try:
+            exceeding_tx = self.nodes[1].sc_send(forwardTransferOuts * 2, {"fee": Decimal(0.0001 * 5), "minconf": 0})
+        except JSONRPCException as e:
+            print(e.error['message'])
+            assert(False)
+        mark_logs(f"Sent transaction FT #4095 and #4096", self.nodes, DEBUG_MODE, 'c')
+        mark_logs(f"  4 - {exceeding_tx}", self.nodes, DEBUG_MODE, 'e')
+        self.sync_all()
+
+        # Ask for a new block template
+        time.sleep(5)
```

Comment:
Can we get rid of this `sleep` now that we have the `wait_for_block_template()` function?

```suggestion
```

Reply:
Addressed in 9ce2a8aff3